chirp/drivers/icm710.py

```diff
@@ -0,0 +1,178 @@
+    _highbit_flip = False
+
+    # Return information about this radio's features, including
+    # how many memories it has, what bands it supports, etc
+    def get_features(self):
+        rf = chirp_common.RadioFeatures()
+        rf.has_bank_index = False
+        rf.has_dtcs = False
+        rf.has_rx_dtcs = False
+        rf.has_dtcs_polarity = False
+        rf.has_mode = True
+        rf.has_offset = False
+        rf.has_name = True
+        rf.has_bank = False
+        rf.has_bank_names = False
+        rf.has_tuning_step = False
+        rf.has_ctone = False
+        rf.has_cross = True
+        rf.has_infinite_number = False
+        rf.has_nostep_tuning = True
+        rf.has_comment = True
+        rf.has_settings = False  # TODO radio does have settings
+        rf.has_variable_power = False
+
+        rf.memory_bounds = (0, 231)
+        rf.valid_bands = [(1000000, 29999999)]
+
+        rf.valid_duplexes = DUPLEX_MODES
+        rf.can_odd_split = True
+        rf.valid_modes = MODULATION_MODES
+        rf.valid_power_levels = POWER_LEVELS
+        rf.valid_tones = []
+        rf.valid_tmodes = []
+        rf.valid_skips = []
+        return rf
+
+    # Convert the raw byte array into a memory object structure
+    def process_mmap(self):
+        self._memobj = bitwise.parse(MEM_FORMAT, self._mmap)
+
+    # Return a raw representation of the memory object, which
+    # is very helpful for development
+    def get_raw_memory(self, number):
+        return repr(self._memobj.memory[number])
+
+    def sync_out(self):
+        self.pipe.baudrate = 4800
+        icf.clone_to_radio(self)
+
+    # Extract a high-level memory object from the
+    # low-level memory map
+    # This is called to populate a memory in the UI
+    def get_memory(self, number):
+
+        # this just gets the 160 common
+        # the 72 ITU channels are probably working
+        # there may be some strange behavior with
+        # the SITOR FSK "hidden" channels
+        # best be careful about trying to set a fixed
+        # simplex channel with a duplex frequency
+        # TODO test, test, test, all the combinations.
+
+        # Get a low-level memory object mapped to the image
+        _mem = self._memobj.memory[int(number)]
+
+        # Create a high-level memory object to return to the UI
+        mem = chirp_common.Memory()
+
+        mem.number = number
+        mem.comment = ''
+
+        # Convert your low-level frequency to Hertz.
+        mem.freq = int(_mem.rx_freq)
+        mem.name = str(_mem.name).rstrip()  # Set the alpha tag
+
+        mem.mode = MODULATION_MODES[0]
+        if (_mem.mode > -1) and (_mem.mode < 7):
+            mem.mode = MODULATION_MODES[int(_mem.mode)]
+
+        # if the frequency is above 30MHz
+        # (166666665 or FF,FF,FF,FF), it's simplex
+        mem.offset = int(_mem.tx_freq)
+        if int(_mem.tx_freq) < 30000000:
+            mem.duplex = 'split'
+            mem.comment = 'TX: ' + \
```

Comment:
The comment is for the user, you're not allowed to coerce it to some value you want it to be. Some drivers can store the comment themselves, but since you can't chirp will store the comment in the metadata when it saves a `.img` file. So you should just remove this.

Reply:
I see. Is there a better way to show what the TX frequency is when the radio is in split mode? Maybe I should explore VFO A/B as a way?

Reply:
As I mentioned on the ML, `split` duplex stores the TX frequency in the offset field. So if `tx_freq != rx_freq`, you need to set `mem.offset = tx_freq` and `duplex='split'`. Also, it looks like you're only supporting `duplex='split'` and `duplex='off'` but not 'duplex=''`. The latter means "simplex" (i.e. same TX and RX frequency). The test framework assumes simplex for its test channel, which is one reason you're failing a bunch of tests. Any user familiar with CHIRP will assume they can set memories in simplex mode.

Reply:
Yeah, I've been banging my head on how to get the unit tests to pass.  I think the radio works like this:
if RX frequency is set and TX is FFFFFF, it's receive only, like WWV
if RX = TX, it's simplex
if RX != TX but valid tx, it's duplex.  In reality, all channels are duplex, but I think we can hide that from the user.  In the ICOM software you explicitly have to set both RX and TX frequencies.  Further, some of the global settings will prevent certain memory channels to be forced simplex depending on the setting.  I don't quite know how that works yet. 
So if duplex ='' then set TX = RX; if duplex = 'split' set the offset, and if duplex = 'off' set the field raw to FFFFFFFF

Reply:
Right, this is a very common arrangement in lots of radios. The job of the driver is to adapt between CHIRP's model and the radio one. So in this case, you _must_ act like the radio has a specific simplex mode by just setting the tx and rx freq the same.